src/StopwatchPlugin.php

```diff
@@ -39,11 +40,11 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
         $this->stopwatch->start($eventName, self::CATEGORY);
 
         return $next($request)->then(function (ResponseInterface $response) use ($eventName) {
-            $this->stopwatch->stop($eventName, self::CATEGORY);
+            $event = $this->stopwatch->stop($eventName);
 
-            return $response;
+            return $response->withHeader(self::HEADER, $event->getDuration());
```

Comment:
yep, same here, but i didn't find a better one. i think getting the information out of stopwatch in collector is not good option. i could introduce some kind of request registry which is used by stopwatch plugin and collector, but i think it would harden the coupling, which might not be the best idea as well. so i decided to go with a header and i think if a plugin manipulates the response with an additional header that's not such a weird behavior, but i clearly get your point. did i miss a good option to avoid the header?